clients/src/test/java/org/apache/kafka/common/record/DefaultRecordBatchTest.java

```diff
@@ -153,6 +154,67 @@ public void testInvalidRecordSize() {
     }
 
     @Test(expected = InvalidRecordException.class)
+    public void testInvalidRecordSizeTooLow() {
```

Comment:
Both of these cases are still testing the same thing. I think you are intending to set an invalid record count, but this is actually changing the size of the batch (i.e. the size in bytes). So whether it is 2 or 10, we're validating the same path.